tools/walletextension/vk_repository.go

```diff
@@ -0,0 +1,72 @@
+package walletextension
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto/ecies"
+)
+
+func NewViewingKeyRepository() *ViewingKeyRepository {
+	return &ViewingKeyRepository{
+		viewingKeysPrivate: make(map[common.Address]*ecies.PrivateKey),
+		viewingKeysPublic:  make(map[common.Address][]byte),
+	}
+}
+
+type ViewingKeyRepository struct {
+	viewingKeysPrivate map[common.Address]*ecies.PrivateKey // Maps an address to its private viewing key.
```

Comment:
let's also create a type `ViewingKey` with the fields: - the private key, the pub key, and the signature